server/src/utils/getConfig.ts

```diff
@@ -56,13 +69,43 @@ const getPostCallConfig = (defaultConfig: ServerConfigModel): PostCallConfig | u
 
   try {
     const postcallSurveyType = process.env[VV_POSTCALL_SURVEY_TYPE_ENV_NAME] ?? defaultConfig.postCall?.survey.type;
-    const postcallSurveyUrl =
-      process.env[VV_POSTCALL_SURVEY_OPTIONS_SURVEYURL_ENV_NAME] ?? defaultConfig.postCall?.survey.options?.surveyUrl;
+    let postcallSurveyUrl = '';
+    if (postcallSurveyType === 'msforms') {
+      const options: MSFormsSurveyOptions = defaultConfig.postCall?.survey?.options as MSFormsSurveyOptions;
+      postcallSurveyUrl = process.env[VV_POSTCALL_SURVEY_OPTIONS_SURVEYURL_ENV_NAME] ?? options.surveyUrl;
+    } else if (postcallSurveyType === 'custom') {
+      const options: CustomSurveyOptions = defaultConfig.postCall?.survey?.options as CustomSurveyOptions;
+      postcallSurveyUrl = process.env[VV_POSTCALL_SURVEY_OPTIONS_SURVEYURL_ENV_NAME] ?? options.surveyUrl;
+    } else {
```

Comment:
Making this `else if (postCallSurveyType === 'onequestionpoll')` is better I think. Then inside that block is where we create the 1QP Config object.

Then, at the end is the `else` block where we should return `undefined` which is basically setting postcall to "None". At that point, it means that postCallSurveyType did not match any of `msforms`, `custom` or `onequestionpoll`

Reply:
If it helps make the code more concise/readable, I suggest creating functions that return the right options object based on survey type:

`const createMSFormsOptions = (): MSFormsSurveyOptions => { ... }`
`const createCustomFormOptions = (): CustomSurveyOptions => { ... }`
`const createOneQuestionPollOptions = (): OneQuestionPollOptions => { ... }`

```
if (postcallSurveyType === 'msforms') {
    const options: MSFormsSurveyOptions = createMSFormsOptions();
    return {
        survey: {
            type: 'msforms',
            options: options
        }
    }
}
```

Reply:
Agreed, made changes accordingly.